flyway.go

```diff
@@ -88,66 +99,99 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 	}, nil
 }
 
-func WithEnvUser(user string) testcontainers.CustomizeRequestOption {
+func parseRequest(req testcontainers.GenericContainerRequest) error {
+	// parse migrations
+	const migrationsErrMessage string = "Please use flyway.WithMigrations() option to provide migrations"
+
+	if req.Env[flywayEnvLocationsKey] == "" {
+		return fmt.Errorf("missing migrations: environment variable %s is empty. %s", flywayEnvLocationsKey, migrationsErrMessage)
+	}
+
+	if len(req.Files) == 0 {
+		return fmt.Errorf("missing migrations: no files provided. %s", migrationsErrMessage)
+	} else {
+		migrationsFound := false
+		for _, file := range req.Files {
+			if file.ContainerFilePath == DefaultMigrationsPath {
+				migrationsFound = true
+			}
+		}
+
+		if !migrationsFound {
+			return fmt.Errorf("missing migrations: %s", migrationsErrMessage)
+		}
+	}
+
+	// parse connection settings
+	if req.Env[flywayEnvUrlKey] == "" {
+		return fmt.Errorf("missing database url: environment variable %s is empty", flywayEnvUrlKey)
+	}
+	if req.Env[flywayEnvUserKey] == "" {
+		return fmt.Errorf("missing user: environment variable %s is empty", flywayEnvUserKey)
+	}
+	if req.Env[flywayEnvPasswordKey] == "" {
+		return fmt.Errorf("missing password: environment variable %s is empty", flywayEnvPasswordKey)
+	}
+
+	return nil
+}
+
+func WithUser(user string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_USER", user)
 }
 
-func WithEnvPassword(password string) testcontainers.CustomizeRequestOption {
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
```

Comment:
With these type of funcs I tend to think that the name reflects what you're changing ... by including Env in the name I wanted it to be clear it was an environment setting, again not significant really and happy to go with your choice

Reply:
But the module should hide the implementation details, so users should not care about if it's an env, a config file, or any other thing. We should probably add comments for each option.